### DISCUSSION TITLE
Background blur is not applied to storyboard sprites

### DISCUSSION THREAD

**Discussion by Wieku** [ppy/osu#13001] [Ideas]
Only the background (if it's not mentioned in the storyboard file) is blurred out:
![osu_2018-10-23_18-18-12](https://user-images.githubusercontent.com/5307179/47375546-d0e58d00-d6f0-11e8-83b5-10407f1f8cfc.png)

**Comment by smoogipoo** [comment]
Yeah idk if we want to apply the blur to the storyboard - it's very computationally expensive, at least with the current implementation of the blur.

Marking this as a low-prio proposal.

**Reply by JustSimplyKyle** [reply]
Is the current implmentation fast gaussian blur"?

**Reply by peppy** [reply]
You can find the implementation here https://github.com/ppy/osu-framework/blob/master/osu.Framework/Resources/Shaders/sh_Blur.fs

**Comment by WebFreak001** [comment]
if this does get implemented, can we make it a toggleable option? When storyboards use these sprites creatively it looks way cooler than when just blurring everything in the background. My favorite map example for this is [Secret Sister Complex](https://osu.ppy.sh/beatmapsets/683123#osu/1444708) which has pretty cool flashing of the unblurred background for a few beats/transitions. Even though it wasn't designed for that I think you could do pretty cool designs with just the background being blurred.

Or what would be even cooler for mapping is storing some flag somewhere which sprites and textures to blur. (and also if they should blur over the edge or if they fill the screen) Then generic things could be blurred but things that should always be sharp such as lyrics/other texts or small light effects could stay sharp.
